nupic/research/frameworks/vernon/mixins/prune_low_snr.py

```diff
@@ -116,9 +109,90 @@ def run_epoch(self):
     @classmethod
     def get_execution_order(cls):
         eo = super().get_execution_order()
-        eo["setup_experiment"].append("PruneLowSNR: Initialize")
-        eo["pre_epoch"].append("PruneLowSNR: Maybe prune")
-        eo["run_epoch"].append("PruneLowSNR: Log parameter sparsity")
+        eo["setup_experiment"].append("PruneLowSNRLayers: Initialize")
+        eo["pre_epoch"].append("PruneLowSNRLayers: Maybe prune")
+        eo["run_epoch"].append("PruneLowSNRLayers: Log parameter sparsity")
+        return eo
+
+
+class PruneLowSNRGlobal:
+    """
+    Prunes weights by signal-to-noise ratio across the entire model. Requires
+    variational dropout modules. For this mixin to work, the model must have a
+    VDropCentralData module stored in the model as self.vdrop_central_data.
+    """
+
+    def setup_experiment(self, config):
+        """
+        :param config:
+            - prune_schedule: A list of (epoch, sparsity) pairs.
+            - validate_on_prune: Whether to run validation after pruning.
+        """
+        super().setup_experiment(config)
+        self.prune_schedule = dict(config["prune_schedule"])
+        self.validate_on_prune = config.get("validate_on_prune", False)
+        self.log_module_sparsities = config.get("log_module_sparsities", False)
+
+    def pre_epoch(self):
+        super().pre_epoch()
+        if self.current_epoch in self.prune_schedule:
+            target_density = self.prune_schedule[self.current_epoch]
+            vdrop_data = self.model.vdrop_central_data
+
+            z_logalpha = vdrop_data.compute_z_logalpha()
+            z_mask = vdrop_data.z_mask
+            z_mu = vdrop_data.z_mu
+            z_logvar = vdrop_data.z_logvar
+
+            num_weights = math.floor(z_logalpha.numel() * target_density)
+            on_indices = z_logalpha.topk(num_weights, largest=False)[1]
+
+            z_mask.zero_()
+            z_mask[on_indices] = 1
+            z_mu.data *= z_mask
+            with torch.no_grad():
+                z_logvar[~z_mask.bool()] = vdrop_data.pruned_logvar_sentinel
+
+            if not self.logger.disabled:
```

Comment:
This check isn't necessary. It's already built into `logger.info`. We only need to do this type of check if we need to compute something expensive before logging.